lib/tessel/provision.js

```diff
@@ -170,6 +180,27 @@ actions.checkAuthFileExists = function(tessel, authFile) {
   });
 };
 
+actions.setDefaultKey = function(opts) {
+
+  if (opts.key && typeof opts.key === 'string') {
+    debug('Using --key ' + opts.key + ' after checking existens');
+    filepath = opts.key;
+  } else if (opts.key && typeof opts.key !== 'string') {
+    logs.err('No valid path for private key');
```

Comment:
Sorry to go back and forth on this one but we should have a consistent way of handling errors. In the rest of the CLI, the error is handled by calling `reject` in a promise with an error. I know I originally suggested this function by synchronous but in an effort to keep the error handling process consistent, let's move this function to a Promise and use `resolve` and `reject`.